kubernetes/istio.go

```diff
@@ -328,13 +328,13 @@ func DestinationRuleHasMTLSEnabled(destinationRule *networking_v1.DestinationRul
 func ClusterNameFromIstiod(conf *config.Config, k8s ClientInterface) (string, error) {
 	// The "cluster_id" is set in an environment variable of
 	// the "istiod" deployment. Let's try to fetch it.
-	istiodDeployments, err := k8s.GetDeployments(conf.ExternalServices.Istio.RootNamespace, metav1.ListOptions{LabelSelector: "app=istiod"})
+	istiodDeployments, err := k8s.GetDeployments("", meta_v1.ListOptions{LabelSelector: "app=istiod"})
```

Comment:
You'll need some special handling here for when Kiali doesn't have cluster wide permissions otherwise this will fail when there's discovery selectors defined for Kiali.

Reply:
Hmm, I'm not exactly sure what the special handling should be. Why would it fail against all-namespaces?

Reply:
With these changes it will be required for the discovery selectors to include istiod namespaces. There is a doco PR that adds this info: https://github.com/kiali/kiali.io/pull/907/files.